src/rtk/dataSlice.ts

```diff
@@ -0,0 +1,37 @@
+import { ActionCreatorWithoutPayload, createSlice, PayloadAction, Slice, SliceCaseReducers } from '@reduxjs/toolkit';
+
+interface DataState {
+  isDay: boolean;
+  editorText: string;
+  responseText: string;
+}
+
+const initialState: DataState = {
+  isDay: true,
+  editorText: '',
+  responseText: 'There will be a response'
+};
+
+export const dataSlice: Slice<DataState, SliceCaseReducers<DataState>, 'somename'> = createSlice({
+  name: 'somename',
+  initialState,
+  reducers: {
+    setIsDay(state, action: PayloadAction<boolean>) {
+      state.isDay = action.payload;
+      if (!state.isDay) document.body.classList.add('dark');
+      if (state.isDay) document.body.classList.remove('dark');
```

Comment:
rely on action payload, not on state property here

also please remove functions that do this from the component